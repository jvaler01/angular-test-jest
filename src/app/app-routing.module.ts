import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { CounterComponent } from './basic/counter/counter.component';
import { PokemonComponent } from './basic/pokemon/pokemon.component';
import { FatherComponent } from './basic/father/father.component';
import { CounterRouteComponent } from './basic/counter-route/counter-route.component';

const routes: Routes = [
  {
    path: 'basic/counter',
    component: CounterComponent
  },
  {
    path: 'basic/counter/:initial',
    component: CounterRouteComponent
  },
  {
    path: 'basic/pokemon',
    component: PokemonComponent
  },
  {
    path: 'basic/father',
    component: FatherComponent
  },
  {
    path: '**',
    redirectTo: 'basic/counter'
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
