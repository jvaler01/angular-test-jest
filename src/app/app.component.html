<div class="container">
    <h1>{{title}}</h1>
    <a class="btn btn-dark mx-1" [routerLink]="['/basic/counter']" routerLinkActive="router-link-active">Counter</a>
    <a class="btn btn-dark mx-1" [routerLink]="['/basic/counter/20']" routerLinkActive="router-link-active">Counter Route</a>
    <a class="btn btn-dark mx-1" [routerLink]="['/basic/pokemon']" routerLinkActive="router-link-active">Pokemon</a>
    <a class="btn btn-dark mx-1" [routerLink]="['/basic/father']" routerLinkActive="router-link-active">Father</a>

    <router-outlet></router-outlet>
</div>
