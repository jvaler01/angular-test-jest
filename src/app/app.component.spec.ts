import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent
      ],
    }).compileComponents();
  });

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    expect(component).toBeTruthy();
  });

  it(`should have as title 'angular-testing'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    expect(component.title).toEqual('angular-testing');
  });

  it('should render title', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;

    const h1 = compiled.querySelector('h1');
    expect( h1?.textContent ).toContain( component.title );
    //expect(compiled.querySelector('.content span')?.textContent).toContain('angular-testing app is running!');
  });

  it('check snapshoot', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;
    expect( compiled ).toMatchSnapshot();
  });
});
