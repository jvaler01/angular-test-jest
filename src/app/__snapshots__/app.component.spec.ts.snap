// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent check snapshoot 1`] = `
<div
  id="root3"
>
  <h1>
    angular-testing
  </h1>
  <router-outlet />
  
</div>
`;
